File: src/main/scala/resonantinduction/mechanical/turbine/TurbineMBlockHandler.java
package resonantinduction.mechanical.turbine;

import net.minecraft.tileentity.TileEntity;
import resonant.lib.multiblock.reference.MultiBlockHandler;
import universalelectricity.core.transform.vector.Vector3;

public class TurbineMBlockHandler extends MultiBlockHandler<TileTurbine>
{
    public TurbineMBlockHandler(TileTurbine wrapper)
    {
        super(wrapper);
    }

    public TileTurbine getWrapperAt(Vector3 position)
    {
        TileEntity tile = position.getTileEntity(this.tile.getWorld());

        if (tile != null && wrapperClass.isAssignableFrom(tile.getClass()))
        {
            if (((TileTurbine) tile).getDirection() == this.tile.getDirection() && ((TileTurbine) tile).tier == this.tile.tier)
            {
                return (TileTurbine) tile;
            }
        }

        return null;
    }
}

File: src/main/scala/resonantinduction/mechanical/mech/turbine/TurbineMBlockHandler.scala
package resonantinduction.mechanical.mech.turbine

import net.minecraft.tileentity.TileEntity
import resonant.lib.prefab.multiblock.reference.MultiBlockHandler
import resonant.lib.transform.vector.Vector3

class TurbineMBlockHandler(wrapper: TileTurbine) extends MultiBlockHandler[TileTurbine](wrapper)
{
    override def getWrapperAt(position: Vector3): TileTurbine =
    {
        val tile: TileEntity = position.getTileEntity(this.tile.getWorld)
        if (tile != null && wrapperClass.isAssignableFrom(tile.getClass))
        {
            if (tile.asInstanceOf[TileTurbine].getDirection == this.tile.getDirection && tile.asInstanceOf[TileTurbine].tier == this.tile.tier)
            {
                return tile.asInstanceOf[TileTurbine]
            }
        }
        return null
    }
}